flusher.go

```diff
@@ -19,57 +19,122 @@ import (
 // Flush takes the slices of metrics, combines then and marshals them to json
 // for posting to Datadog.
 func (s *Server) Flush(interval time.Duration, metricLimit int) {
+	// right now we have only one destination plugin
+	// but eventually, this is where we would loop over our supported
+	// destinations
+	if s.IsLocal() {
+		s.FlushLocal(interval, metricLimit)
+	} else {
+		s.FlushGlobal(interval, metricLimit)
+	}
+}
+
+func (s *Server) FlushGlobal(interval time.Duration, metricLimit int) {
 	go s.flushEventsChecks() // we can do all of this separately
 
 	percentiles := s.HistogramPercentiles
-	if s.ForwardAddr != "" {
-		// don't publish percentiles if we're a local veneur; that's the global
-		// veneur's job
-		percentiles = nil
-	}
 
+	tempMetrics, ms := s.tallyMetrics(percentiles)
+
+	// the global veneur instance is also responsible for reporting the sets
+	ms.totalLength += ms.totalSets
+
+	finalMetrics := s.generateDDMetrics(interval, percentiles, tempMetrics, ms)
+
+	s.reportMetricsFlushCounts(ms)
+
+	s.reportGlobalMetricsFlushCounts(ms)
+
+	s.flushRemote(finalMetrics, metricLimit)
```

Comment:
I can remove the linebreaks.

The reason we're avoiding the use of the word `Datadog` specifically is that we shouldn't *really* care where they're going - we only want to say `Datadog` when we're doing something that's truly DataDog-specific. (This isn't currently the case with our naming scheme; plenty of older parts of the codebase use `dd` and `datadog`, but we want to move in that direction.